### DISCUSSION TITLE
so, when will there be a vite plugin for stylex?

### DISCUSSION THREAD

**Discussion by ybf970928** [facebook/stylex#52] [General]
need like @vitejs/plugin-stylex

**Comment by nmn** [comment]
TLDR; 

Making things work with `vite build` are fairly easy. `vite-plugin-stylex` was published by a [member of the community](https://github.com/facebook/stylex/issues/47#issuecomment-1842174599) which looks fairly similar to the Rollup plugin.

But, we don't want to rely on runtime style injection, even in development, to be able to support RSCs and Islands Architecture. This has been challenging with Vite's unbundled dev server.

I'm hoping someone from the Vite community will help us solve this problem.

**Comment by nmn** [comment]
I worked with the the Qwik.js team and we have this working pretty well:

https://github.com/nmn/qwik-stylex

It should be possible to generalise this for most (all?) Vite apps.
